packages/relay/src/lib/services/cacheService/cacheService.ts

```diff
@@ -91,12 +91,13 @@ export class CacheService {
     this.register = register;
 
     this.internalCache = new LocalLRUCache(logger.child({ name: 'localLRUCache' }), register);
```

Comment:
QQ: Is this `LocalLRUCache` object ever going to be used if `this.isSharedCacheEnabled === true`? 

We can instantiate it like this to avoid wasting heap space:

```diff
-    this.internalCache = new LocalLRUCache(logger.child({ name: 'localLRUCache' }), register);
     ...
     if (this.isSharedCacheEnabled) {
       this.sharedCache = new RedisCache(logger.child({ name: 'redisCache' }), register);
     } else {
-      this.sharedCache = this.internalCache;
+      this.sharedCache = new LocalLRUCache(logger.child({ name: 'localLRUCache' }), register);
     }
```

Reply:
If I introduce the change like this typescript will throw an error when building, since it doesn't allow conditional initialisation in the constructor.
An option is to add non-null assertion at the declaration - like this:
`/**
   * The LRU cache used for caching items from requests.
   *
   * @private
   */
  private readonly internalCache!: ICacheClient;

  /**
   * The Redis cache used for caching items from requests.
   *
   * @private
   */
  private readonly sharedCache!: ICacheClient;`
  Are you ok with this?

Reply:
If I introduce the change like this typescript will throw an error when building, since it doesn't allow conditional initialisation in the constructor.
An option is to add non-null assertion at the declaration - like this:
```
/**
   * The LRU cache used for caching items from requests.
   *
   * @private
   */
  private readonly internalCache!: ICacheClient;

  /**
   * The Redis cache used for caching items from requests.
   *
   * @private
   */
  private readonly sharedCache!: ICacheClient;
```
What do you think about this?

Reply:
I think we should use only one variable instead of two:
```
  /**
   * The client used for caching items from requests.
   *
   * @private
   */
  private readonly cacheClient: ICacheClient;

  constructor() {
    ...
    if (this.isSharedCacheEnabled) {
      this.cacheClient = new RedisCache(logger.child({ name: 'redisCache' }), register);
    } else {
      this.cacheClient = new LocalLRUCache(logger.child({ name: 'localLRUCache' }), register);
    }
  }
```

It's unnecessary to have two variables if we are only going to use one of them.

Reply:
Okay, I just saw that we use the internal cache as fallback in case we get errors from redis.

Disregard the comments above. Let's revert this to as it was before - always initializing the LRU client and optionally initializing the redis client based on `this.isSharedCacheEnabled`

Reply:
Yeah, you are right, we are supposed to fallback to internalCache, but in some methods I think we dont actually do that
Will probably open an issue about it